ngraph/frontend/onnx_import/src/core/graph.cpp

```diff
@@ -272,7 +273,21 @@ namespace ngraph
                     break;
                 }
 
-                ng_node_vector[i].get_node()->set_friendly_name(onnx_node.output(i));
+                auto onnx_node_name = onnx_node.get_name();
+                if (onnx_node_name.empty())
+                {
+                    ng_node_vector[i].get_node()->set_friendly_name(onnx_node.output(i));
+                }
+                else
+                {
+                    ng_node_vector[i].get_node()->set_friendly_name(onnx_node.get_name());
+                }
+
+                // null node does not have tensor
+                if (!ngraph::op::is_null(ng_node_vector[i]))
```

Comment:
Should we add a check that node is not a result? Or don't we have result operations here?

Reply:
ONNX Importer does not produce result operations.
This type of nodes are added by ngraph function when it is created.